plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/connection/OAuthDetailView.java

```diff
@@ -145,11 +145,21 @@ public Composite createControls(Composite parent, Object context, DataBindingCon
 	public void onVisible(IObservableValue detailsViewModel, DataBindingContext dbc) {
 		bindWidgetsToInternalModel(dbc);
 		rememberTokenCheckbox.setText("&Save token (could trigger secure storage login)");
+		setCheckboxVisible(true);
+		rememberTokenCheckbox.getParent().update();
+		rememberTokenCheckbox.getParent().layout(true);
 	}
 	
 	@Override
 	public void onInVisible(IObservableValue detailsViewModel, DataBindingContext dbc) {
 		dispose();
+		setCheckboxVisible(false);
+		((GridData)rememberTokenCheckbox.getLayoutData()).heightHint = 1;
```

Comment:
It is a good utillity, but in this case we have two change visibility of two checkboxes but it is better to update layout once. Doing it for each checkbox will make visible jumps in dialog. I did not know about exclude, sure it must be better than height hint.